### PR TITLE
migrate scheduler/internal logs into structured logging

### PR DESCRIPTION
**Pull request by CKchen0726**
What type of PR is this?
/kind cleanup

What this PR does / why we need it:
part of kubernetes/enhancements#1602

Which issue(s) this PR fixes:
keps/sig-instrumentation/1602-structured-logging
Structured Logging migration instructions

Special notes for your reviewer:

Does this PR introduce a user-facing change?:
```release-note
migrate scheduler/internal logs into structured logging
```

### CODE REVIEW

File: pkg/scheduler/internal/cache/cache.go [hunk 1]
```diff
@@ -525,7 +525,7 @@ func (cache *schedulerCache) UpdatePod(oldPod, newPod *v1.Pod) error {
 	// before Update event, in which case the state would change from Assumed to Added.
 	case ok && !cache.assumedPods[key]:
 		if currState.pod.Spec.NodeName != newPod.Spec.NodeName {
-			klog.Errorf("Pod %v updated on a different node than previously added to.", key)
+			klog.ErrorS(nil, "Pod updated on a different node than previously added to.", "podUID", key)
```
[the remark this answers is not in the document]
Comment (CKchen0726) [reply pkg/scheduler/internal/cache/cache.go]: Thanks for your suggestion, I have updated it

File: pkg/scheduler/internal/cache/cache.go [hunk 2]
```diff
@@ -459,7 +459,7 @@ func (cache *schedulerCache) updatePod(oldPod, newPod *v1.Pod) error {
 func (cache *schedulerCache) removePod(pod *v1.Pod) error {
 	n, ok := cache.nodes[pod.Spec.NodeName]
 	if !ok {
-		klog.Errorf("node %v not found when trying to remove pod %v", pod.Spec.NodeName, pod.Name)
+		klog.ErrorS(nil, "node not found when trying to remove pod", "nodeName", pod.Spec.NodeName, "podName", pod.Name)
```
[the remark this answers is not in the document]
Comment (CKchen0726) [reply pkg/scheduler/internal/cache/cache.go]: updated

File: pkg/scheduler/internal/cache/cache.go [hunk 3]
```diff
@@ -125,7 +125,7 @@ func newNodeInfoListItem(ni *framework.NodeInfo) *nodeInfoListItem {
 func (cache *schedulerCache) moveNodeInfoToHead(name string) {
 	ni, ok := cache.nodes[name]
 	if !ok {
-		klog.Errorf("No NodeInfo with name %v found in the cache", name)
+		klog.ErrorS(nil, "No NodeInfo with nodeName found in the cache", "nodeName", name)
```
[the remark this answers is not in the document]
Comment (CKchen0726) [reply pkg/scheduler/internal/cache/cache.go]: Thanks, I have updated it.

**Comment by Huang-Wei** [comment]
Is it overlapped with #97612?

File: pkg/scheduler/internal/cache/cache.go [hunk 4]
```diff
@@ -525,8 +525,7 @@ func (cache *schedulerCache) UpdatePod(oldPod, newPod *v1.Pod) error {
 	// before Update event, in which case the state would change from Assumed to Added.
 	case ok && !cache.assumedPods[key]:
 		if currState.pod.Spec.NodeName != newPod.Spec.NodeName {
-			klog.Errorf("Pod %v updated on a different node than previously added to.", key)
-			klog.Fatalf("Schedulercache is corrupted and can badly affect scheduling decisions")
+			klog.ErrorS(fmt.Errorf("schedulercache is corrupted and can badly affect scheduling decisions"), "Pod updated on a different node than previously added to", "pod", key)
```
Comment (yangjunmyfm192085) [inline pkg/scheduler/internal/cache/cache.go:528]: Maybe missed os.Exit(1)

Comment (CKchen0726) [reply pkg/scheduler/internal/cache/cache.go:528]: Yes, It was poor thinking-through of a code change. I will fix it.

**Comment by Huang-Wei** [comment]
@ahg-g @damemi could you help review this as you already reviewed on a similar (but now closed) PR (#97612).

File: pkg/scheduler/internal/cache/cache.go [hunk 5]
```diff
@@ -733,17 +734,17 @@ func (cache *schedulerCache) cleanupAssumedPods(now time.Time) {
...
-				ps.pod.Namespace, ps.pod.Name)
+			klog.V(5).InfoS("Couldn't expire cache for pod. Binding is still in progress", "pod", klog.KObj(ps.pod))
 			continue
 		}
 		if now.After(*ps.deadline) {
-			klog.Warningf("Pod %s/%s expired", ps.pod.Namespace, ps.pod.Name)
+			klog.InfoS("Pod in specified namespace is expired", "namespace", ps.pod.Namespace, "pod", ps.pod.Name)
 			if err := cache.expirePod(key, ps); err != nil {
-				klog.Errorf("ExpirePod failed for %s: %v", key, err)
+				klog.ErrorS(err, "ExpirePod failed for pod", "pod", key)
```
Comment (ahg-g) [inline pkg/scheduler/internal/cache/cache.go:747]: use klog.KObj(pod) instead of key

File: pkg/scheduler/internal/cache/cache.go [hunk 6]
```diff
@@ -733,17 +734,17 @@ func (cache *schedulerCache) cleanupAssumedPods(now time.Time) {
...
 		}
 		if !ps.bindingFinished {
-			klog.V(5).Infof("Couldn't expire cache for pod %v/%v. Binding is still in progress.",
-				ps.pod.Namespace, ps.pod.Name)
+			klog.V(5).InfoS("Couldn't expire cache for pod. Binding is still in progress", "pod", klog.KObj(ps.pod))
 			continue
 		}
 		if now.After(*ps.deadline) {
-			klog.Warningf("Pod %s/%s expired", ps.pod.Namespace, ps.pod.Name)
+			klog.InfoS("Pod in specified namespace is expired", "namespace", ps.pod.Namespace, "pod", ps.pod.Name)
```
Comment (ahg-g) [inline pkg/scheduler/internal/cache/cache.go:745]: use klog.KObj(pod) instead

File: pkg/scheduler/internal/cache/cache.go [hunk 7]
```diff
@@ -520,8 +521,8 @@ func (cache *schedulerCache) UpdatePod(oldPod, newPod *v1.Pod) error {
 	// before Update event, in which case the state would change from Assumed to Added.
 	case ok && !cache.assumedPods.Has(key):
 		if currState.pod.Spec.NodeName != newPod.Spec.NodeName {
-			klog.Errorf("Pod %v updated on a different node than previously added to.", key)
-			klog.Fatalf("Schedulercache is corrupted and can badly affect scheduling decisions")
+			klog.ErrorS(fmt.Errorf("schedulercache is corrupted and can badly affect scheduling decisions"), "Pod updated on a different node than previously added to", "pod", key)
+			os.Exit(1)
```
Comment (ahg-g) [inline pkg/scheduler/internal/cache/cache.go:525]: @serathius what is the replacement for `Fatal`?

Comment (Huang-Wei) [reply pkg/scheduler/internal/cache/cache.go:525]: It's `klog.ErrorS()` and `os.Exit(<exit code>)` so the exit code can be customized.

Ref: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#replacing-fatal-calls

**Comment by Huang-Wei** [comment]
@CKchen0726 are you still working on this? If yes, can you address the comments and also migrate the klog usage in `pkg/scheduler/internal/cache/debugger/comparer.go`, so that we can close #101831 as a duplicate as this PR was raised earlier.

**Pull request closed without merging**